extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -121,15 +149,15 @@ export async function createEditAddingLinksForUriList(document: vscode.TextDocum
 	return { additionalEdits, label };
 }
 
-export function checkSmartPaste(documentText: string, start: number, end: number): SmartPaste {
+export function checkSmartPaste(document: SkinnyTextDocument, selectedRange: vscode.Range): SmartPaste {
 	const SmartPaste: SmartPaste = { pasteAsMarkdownLink: true, updateTitle: false };
 	for (const regex of smartPasteRegexes) {
-		const matches = [...documentText.matchAll(regex.regex)];
+		const matches = regex.isInline ? [...document.lineAt(selectedRange.start).text.matchAll(regex.regex)] : [...document.getText().matchAll(regex.regex)];
 		for (const match of matches) {
 			if (match.index !== undefined) {
-				const useDefaultPaste = start > match.index && end < match.index + match[0].length;
+				const useDefaultPaste = selectedRange.start.character > match.index && selectedRange.end.character < match.index + match[0].length;
```

Comment:
Does this work correctly? It seems suspicious because `selectedRange.start.character` is a column number whereas `match.index` is the offset in the full text. See if you can break this in manual testing and then write some unit tests too

Also maybe you can use some of the position/range helpers, such as `Range.contains` or `Range.intersection`

Reply:
Before when I had, each regex (whether inline of not) being matched with the whole document, this was working correctly, because match.index would give the starting position of the match in the whole document text (which can be compared to the offset in the full text). For now, let me revert it back to checking the whole document, and I can work on making it so the inline regexes only and accurately match a line of text.